server/migrations/migrate_v12.js

```diff
@@ -0,0 +1,176 @@
+import { MinutesCollection } from '/imports/collections/minutes_private';
+import { MeetingSeriesCollection } from '/imports/collections/meetingseries_private';
+
+function saveSeries(series) {
+    MeetingSeriesCollection.update(
+        series._id,
+        {
+            $set: {
+                "topics": series.topics,
+                "openTopics": series.openTopics
+            }
+        },
+        { bypassCollection2: true }
+    );
+}
+
+function saveMinutes(minutes) {
+    // We switch off bypassCollection2 here, to skip .clean & .validate to allow empty string values
+    MinutesCollection.update(
+        minutes._id,
+        {
+            $set: {
+                "topics": minutes.topics,
+            }
+        },
+        { bypassCollection2: true }
+    );
+}
+
+
+class MigrateSeriesUp {
+    constructor(series) {
+        this.topicParentMinuteMap = {};
+        this.series = series;
+    }
+
+    run() {
+        let minutes = this.series.firstMinutes();
+        while (minutes) {
+                if (minutes == this.series.firstMinutes()) {
```

Comment:
Please refactor this whole block inside the `while (minutes)`-loop because such a nested structure is not readable anymore. You can do this by moving the nested blocks into separate private methods.

A first step could be:
```
// ...
    while (minutes) {
        let prevMinutes = minutes.previousMinutes();
        if (prevMinutes) {
            this._updatePreviousCreatedTopicItemDetails(minutes, prevMinutes);
        }
        minutes = this._updateTopicsOfMinutes(minutes);
        saveMinutes(minutes);
        minutes = minutes.nextMinutes();
    }
// ....
_updatePreviousCreatedTopicItemDetails(minutes, prevMinutes) {
    minutes.topics.forEach(topic => {
        this._updatePreviousCreatedItemDetails(topic, prevMinutes.topics, minutes._id);
});

_updatePreviousCreatedItemDetails(topic, prevTopics, currentMinutesId) {
    if (!prevTopics) return;
    let prevTopic = prevTopics.find(prevTopic => topic._id === prevTopic._id);
    if (!prevTopic) return;
    topic.infoItems.forEach(infoItem => {
        // call the next private method here and pass necessary parameters
    }
}

```